pkg/controller/registry/registry_controller_test.go

```diff
@@ -59,29 +61,69 @@ func TestReconcile(t *testing.T) {
 	g.Expect(addRegController(mgr, recFn)).NotTo(gomega.HaveOccurred())
 	defer close(StartTestManager(mgr, g))
 
+	// Create Secret
+	secret, err := test.BuildSecretFromCert("foo-ca-crt", "foo.crt")
+
+	if err != nil {
+		t.Errorf("Error creating secret %v", err)
+	}
+
+	err = c.Create(context.TODO(), secret)
+
+	if apierrors.IsInvalid(err) {
+		t.Errorf("failed to create secret: %v", err)
+	}
+
 	// Create the Registry object and expect the Reconcile and Deployment to be created
+	instance, err := kubicv1beta1.GetTestRegistry("foo")
+	if err != nil {
+		t.Errorf("Error Getting Registry %v", err)
+	}
 	err = c.Create(context.TODO(), instance)
+
 	// The instance object may not be a valid object because it might be missing some required fields.
 	// Please modify the instance object by adding required fields and then remove the following if statement.
 	if apierrors.IsInvalid(err) {
-		t.Logf("failed to create object, got an invalid object error: %v", err)
-		return
+		t.Errorf("failed to create registry: %v", err)
 	}
 	g.Expect(err).NotTo(gomega.HaveOccurred())
-	defer c.Delete(context.TODO(), instance)
+
+	defer func() {
+
+		registryName := types.NamespacedName{Name: "foo"}
+		created := &kubicv1beta1.Registry{}
+
+		err0 := c.Get(context.TODO(), registryName, created)
+		if err0 == nil {
```

Comment:
we should also check if the err != nil ( but in defer is not possible so we need to move out)

Reply:
also `err0` is not usual. we should call it `err`